File: src/java/org/javaturk/dp/ch05/adapter/electricity/powerAdapter1/USTurkishPowerAdapter.java
package org.javaturk.dp.ch05.adapter.electricity.powerAdapter1;

import org.javaturk.dp.ch05.adapter.electricity.domain.tr.TurkishPowerSource;
import org.javaturk.dp.ch05.adapter.electricity.domain.us.USPowerSource;

public class USTurkishPowerAdapter implements TurkishPowerSource {
	private USPowerSource usPowerSource;
	private boolean on;

	public USTurkishPowerAdapter(USPowerSource usPowerSource) {
		System.out.println("\nUSTurkishPowerAdapter: Converting from USPowerSource to TurkishPowerSource");
		this.usPowerSource = usPowerSource;
//		convert110To220(usPowerSource);
	}

	@Override
	public void providePowerAt220V() {
		usPowerSource.providePowerAt110V();
		convert110To220(usPowerSource);
	}

	@Override
	public void turnOn() {
		if (!on) {
			usPowerSource.pushSwitch();
			on = true;
		}
	}

	@Override
	public void turnOff() {
		if (on) {
			usPowerSource.pushSwitch();
			on = false;
		}
	}

	private void convert110To220(USPowerSource usPowerSource) {
		System.out.println("USTurkishPowerAdapter: Converting from 110V to 220V");
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/adapter/electricity/powerAdapter1/us-turkish-power-adapter.ts
import {TurkishPowerSource} from "../domain/tr/turkish-power-source";
import {USPowerSource} from "../domain/us/us-power-source";

export class USTurkishPowerAdapter implements TurkishPowerSource {
  private on: boolean;
  private usPowerSource: USPowerSource;

  constructor(usPowerSource: USPowerSource) {
    console.log("\nUSTurkishPowerAdapter: Converting from USPowerSource to TurkishPowerSource");
    this.usPowerSource = usPowerSource;
  }

  providePowerAt220V(): void {
    this.usPowerSource.providePowerAt110V();
    this.convert110To220(this.usPowerSource);

  }

  turnOff(): void {
    if (this.on) {
      this.usPowerSource.pushSwitch();
      this.on = false;
    }
  }

  turnOn(): void {
    if (!this.on) {
      this.usPowerSource.pushSwitch();
      this.on = true;
    }
  }

  private convert110To220(usPowerSource: USPowerSource): void {
    console.log("USTurkishPowerAdapter: Converting from 110V to 220V");
  }
}